aspnet/dnx/commands.rst

```diff
@@ -1,8 +1,121 @@
-.. include:: /../common/stub-topic.txt
+Using Commands
+==============
 
-|stub-icon| Using Commands
-==========================
+DNX projects are used to build and run .NET applications for Windows, Mac and Linux. DNX provides a host process, CLR hosting logic and managed entry point discovery. You can use DNX to execute commands from a command prompt. 
 
-.. include:: /../common/stub-notice.txt
-    
-.. _issue: https://github.com/aspnet/Docs/issues/55
+What is a command?
+------------------
+
+A command is a named execution of a .NET entry point with specific arguments. The *project.json* file allows you to define commands for your projects. Those commands are understood by Visual Studio Code (VS Code) and they will show up in VS Code's **Command Palette**.
+
+Using commands in your project
+------------------------------
+
+In the ``commands`` section of the below *project.json* example, four commands are listed:
+
+.. code-block:: json
+
+    {
+      "version": "1.0.0",
+      "webroot": "wwwroot",
+      "exclude": [
+        "wwwroot"
+      ],
+      "dependencies": {
+        "Kestrel": "1.0.0-beta4",
+        "Microsoft.AspNet.Diagnostics": "1.0.0-beta4",
+        "Microsoft.AspNet.Hosting": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.IIS": "1.0.0-beta4",
+        "Microsoft.AspNet.Server.WebListener": "1.0.0-beta4",
+        "Microsoft.AspNet.StaticFiles": "1.0.0-beta4"
+      },
+      "commands": {
+		"web": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.WebListener --server.urls http://localhost:5000",
+		"kestrel": "Microsoft.AspNet.Hosting --server Kestrel --server.urls http://localhost:5001",
+		"gen": "Microsoft.Framework.CodeGeneration",
+		"ef": "EntityFramework.Commands"
+      },
+      "frameworks": {
+        "dnx451": { },
+        "dnxcore50": { }
+      }
+    }
+
+The commands can include a set of arguments that will be passed to the DNX. In the above example, the first part of a command statement is an assembly with an entry point that the DNX will try to execute. Notice that in the ``commands`` section shown above, the ``ef`` assembly doesn't require any extra arguments, so all that is needed to define the command is the name of the assembly. For the ``kestrel`` command, there are three arguments. The first is the needed assembly, the second is the ``server`` agrument, and the third is the ``server.urls`` argument. The ``kestrel`` command will check the ``Microsoft.AspNet.Hosting`` assembly for an entry point, then it will pass the ``server`` and ``serveral.urls`` parameters to the entry point. 
```

Comment:
> the ``ef`` assembly doesn't require any extra arguments
The `ef` command is implemented by the `EntityFramework.Commands` assembly and doesn't require any extra arguments